social/tasks.py

```diff
@@ -0,0 +1,48 @@
+import feedparser
+from background_task import background
+from social.app.models.author import Author
+from social.app.models.post import Post
+
+# Get the GitHub activity of a user
+# Source: https://pypi.python.org/pypi/django-background-tasks
+@background(schedule=60)
+def get_github_activity(authorId):
+    gitAuthor = Author.objects.get(id=authorId)
+    gitUrl = gitAuthor.github
+
+    # lazy way of checking if URL is correct right now
+    if((gitUrl[:19] == "https://github.com/") and (len(gitUrl.split("/")) == 4)):
```

Comment:
Use a regular expression here. A quick search online yielded examples such as [this one](https://github.com/lorey/social-media-profiles-regexs#github). Please view the following [test cases](https://regex101.com/r/u0w5z2/2).

Also, the second condition here is very fragile and it will only work on URLs that do not terminate in a slash. For example, this GitHub profile page is also valid: https://github.com/davidyee/